kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -204,16 +204,23 @@ public static Map<String, String> validateProperties(Map<String, String> configu
     for (Map.Entry<String, String> kv : configurations.entrySet()) {
       String key = kv.getKey().toLowerCase(Locale.ROOT);
       String value = kv.getValue();
-      if (key.startsWith("delta.") && VALID_PROPERTIES.containsKey(key)) {
+
+      if (key.startsWith("delta.")) {
+        // If it is a delta table property, make sure it is a supported property and editable
+        if (!VALID_PROPERTIES.containsKey(key)) {
+          throw DeltaErrors.unknownConfigurationException(kv.getKey());
+        }
+
         TableConfig<?> tableConfig = VALID_PROPERTIES.get(key);
-        if (tableConfig.editable) {
-          tableConfig.validate(value);
-          validatedConfigurations.put(tableConfig.getKey(), value);
-        } else {
+        if (!tableConfig.editable) {
```

Comment:
could we clarify somewhere, perhaps with a better input parameter name other than `configurations`, that these are new configurations that we are validating (i.e. from a table create/update)

basically -- I was configured why we throw an exception if the tableConfig is not editable

For example, if I read a delta table at version N, got all of the configurations, and passed those configrations into this API --- this would throw an exception, right? but that isn't the intended use case here.

Reply:
updated the method docs

Reply:
Thanks!